pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -202,6 +202,11 @@ func (m *kubeGenericRuntimeManager) startContainer(ctx context.Context, podSandb
 		return msg, err
 	}
 
+	// Don't let the image garbage collection race with container creation
+	// See: https://github.com/kubernetes/kubernetes/issues/123631
+	m.imageGCLock.RLock()
```

Comment:
Will this block the container creation thread? 

Consier a scenario where the image gc is stuck or takes long time to complete and in the meantime we have a lot of container create requests

this lock will block all the container creation threads, will this have any performance impact in terms of the number of threads getting blocked?

Reply:
Yes, we discussed two different appraoches in https://github.com/kubernetes/kubernetes/issues/123631#issuecomment-1983670720. I'm giving it a bit time to wait for @tomergayer's PR.